### ISSUE TITLE
metrics is 0 or 1.0 in training 

### ISSUE THREAD

**Issue by RoyAmoyal** [zju3dv/clean-pvnet#211]
hey, when I train the model on my custom dataset for some reason I get weird numbers: 

![image](https://user-images.githubusercontent.com/69425073/165604721-1d13f7eb-323c-43fb-bb13-1834eaf7efc6.png)

I already tried to re-clone the repository and install everything again but it did not work.

**Note 1: I tried to train the model on 15 images. then I tried again on 1 image for sanity check to see if the model overfitting that image. It doesn't overfiting the image.**

for the 15 training images (with batch-size 3) after 60 epochs, finally the 2d projections metric has changed:
![image](https://user-images.githubusercontent.com/69425073/165621865-3c6b80de-78b1-4caa-9014-30ffbbd63865.png)

So I think the model can't train that well because of the metrics, is that right?
Can you help me understand how to fix that? 

`

@pengsida

**Comment by pengsida** [comment]
Maybe the intrinsic camera matrix causes the problem.

If the code didn't work on our provided custom dataset, I think that it is caused by the pytorch version.

**Comment by LukasMahieuArinti** [comment]
Same issue here. Works fine on the provided LINEMOD dataset, but when I test it on a custom dataset, the mask metric goes to 1 while the other metrics go to 0.
The issue seems to stem from the prediction of the 2d keypoints that come out of the CNN and ransac voting; they just seem to always miss half of my object. All of the projections that come after those steps look fine, but if that first step doesn't work then of course the rest doesn't matter.

I don't think its related to data issues; my data looks exactly like the custom sample and the _train.json_ that is generated makes sense (I mapped the 2d ground truth keypoints to the 3d ones myself; they match up). Also, in that first part of the network where it already goes wrong, the predictions are all still in 2d pixel values, so I don't think there's really all that much that could have gone wrong datawise until then?

Here's a sample prediction of the 2d keypoints my network seems to converge to:
[[319.24655 239.93082]
 [448.37094 269.20236]
 [386.88788 240.74236]
 [299.71332 242.1411 ]
 [373.48035 253.09961]
 [382.12964 214.33127]
 [325.03336 213.0816 ]
 [456.5798  224.25145]
 [313.52698 255.39069]]
 
 While the ground truth is this:
 [[164.58437345373295, 187.17753795578224], [467.446368374397, 286.23647132587666], [395.96128108288445, 243.66007793977536], [249.64268291611103, 211.19976911720144], [420.206730681041, 288.8986210770757], [372.73689584394725, 262.77303680754846], [356.81633824861075, 232.6633354997052], [434.08937542379135, 254.29372235099407]]
 
Close-ish for most points, but that very first point all the way to the left is never found (which omits half the object).
Any ideas why this could happen @pengsida? Something in the CNN, or the ransac voting (or maybe still something in the data that I missed?)

Also, as I always do, I'm testing this on a very small homogenous dataset of very similar poses/pictures in the hope that it would overfit quickly. Maybe that has something to do with it? (even though I don't understand why that wouldn't work in this case)

**Comment by pengsida** [comment]
You could visualize the data to check if it is correct.

**Comment by Vemokaz** [comment]
> I did visualize the data, everything looks correct. I plotted all the 2D points (keypoints and corners) and their matching 3D points as well; they all make sense. Scale shouldn't matter right? (except in the calculation of the ADD metric where the diameter is used)

Hi, I met the same issue, but also visualized the data correctly. I'd like to konw how you deal with the issue finally,thanks!
